[MINOR] Ensure jar loaded for jvm_stdout

gweidner opened the pull request:
Added _get_spark_context call to init of jvm_stdout to prevent potential "TypeError: 'JavaPackage' object is not callable" for edge-cases of very helpful 'with jvm_stdout' python utility.

gweidner commented:
The 'JavaPackage not callable' error was encountered when I used jvm_stdout prior to initializing MLContext.  For example:
`from systemml import MLContext, dml, jvm_stdout
with jvm_stdout():
    ml = MLContext(spark)
    ml.execute(dml("print(\"Hello World!\")"))`
or
`from systemml import MLContext, dml, jvm_stdout, dmlFromResource
with jvm_stdout():
    script = dmlFromResource("/scripts/algorithms/LinearRegCG.dml")`

With the change, the welcome message was visible in notebook similar to when first instantiating in pyspark or spark scala shell.

Welcome to Apache SystemML!
Version 1.0.0-SNAPSHOT
Hello World!
SystemML Statistics:
Total execution time:		0.001 sec.
Number of executed Spark inst:	0.

akchinSTC commented:
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1870/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='SystemML-PullRequestBuilder/org.apache.systemml:systemml' /><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1870/org.apache.systemml$systemml/testReport'>SystemML-PullRequestBuilder/org.apache.systemml:systemml</a>: <span class='status-failure'>1</span></h3><ul><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1870/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.data/FullReblockTest/testBinaryBlockMultipleMSparseSP/'><strong>org.apache.sysml.test.integration.functions.data.FullReblockTest.testBinaryBlockMultipleMSparseSP</strong></a></li></ul>

gweidner commented:
The addition of 'global' keyword at line 55 of mlcontext.py has been addressed with https://github.com/apache/systemml/commit/9da5eab00efa86d1f135664dc7458e5f04c0097e.
But when running with the latest nightly build containing that fix on non-cloud Jupyter environment, I still encountered 'JavaPackage not callable' error as shown in screen capture.

![javapackage_not_callable](https://user-images.githubusercontent.com/12058879/29044487-91a70822-7b74-11e7-98ca-864bf42fda02.png)

Can @niketanpansare review this PR?

niketanpansare commented:
Good catch @gweidner ... Thanks for addressing it ... can you please add a comment in the documentation of `_get_spark_context()` method stating that developers should always use `_get_spark_context()`  rather than `SparkContext._active_spark_context` ? Other than that, LGTM :+1:

gweidner commented:
Good point - thank you @niketanpansare!  I've rebased the branch and added a comment for developers to always use _get_spark_context() in place of SparkContext._active_spark_context.  I will merge.

akchinSTC commented:
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1885/
